### DISCUSSION TITLE
NEW RULE IDEAS

### DISCUSSION THREAD

**Discussion by tcartwright** [tcartwright/SqlServer.Rules#12] [Ideas]
# NEW RULE IDEAS

- [x] not for replication (Design) Added in rule SRD0065
- [ ] Constraints: check for
    - [ ] udf usage (Performance)
    - [ ] nocheck (Design)
- [ ] Suggest SCHEMABINDING for functions that do not touch tables (Performance)
- [ ] Detect sql injection possibilities???

## Notes:
  - Would like to only detect injection opportunies where sql is being concatenated in from a variable.
  - Would also like to build a sql injection scanner for .Net code

**Comment by Fortunoxx** [comment]
- [ ] similar to SRD0016, check for data type inconsistencies in JOIN ON clauses
- [ ] check for data type inconsistencies in INSERT and UPDATE statements (maybe even MERGE)
- prevent data loss when inserting from a source with a higher precedence into a destination with a lower precedence (same goes for UPDATE) 
- the source can be a column, a variable, the result of a function or even the result-set of a stored proc? maybe more...
- example: insert NVARCHAR-data into VARCHAR-column
